Assets/Scripts/SlipTime/SlipTimeManager.cs

```diff
@@ -36,57 +36,111 @@ public class SlipTimeManager : MonoBehaviour
         /// <summary>
         /// The default amount of time it takes to charge one charge of SlipTime.
         /// </summary>
-        public float slipTimeChargeDuration = 3;
+        public float slipTimeCooldownDuration = 60f;
+
+        /// <summary>
+        /// How much SlipTime duration to subtract if the player moves during SlipTime.
+        /// </summary>
+        public float slipTimeMovementPenaltyCoefficient = 0.5f;
+
+        /// <summary>
+        /// The default amount of time to decrease the current cooldown of SlipTime by when bullet grazing.
+        /// </summary>
+        public float bulletGrazingCooldownDecreaseCoefficient = 0.5f;
+
+        /// <summary>
+        /// Counter to keep track of time spent in SlipTime.
+        /// </summary>
+        public float SlipTimeCounter { get; private set; }
+
+        /// <summary>
+        /// Counter to keep track of time spent charging SlipTime.
+        /// </summary>
+        public float ChargeTimeCounter { get; private set; }
+
+        /// <summary>
+        /// Whether the player is in SlipTime.
+        /// </summary>
+        public bool InSlipTime { get; private set; }
 
-        private bool inSlipTime;
-        private double slipTimeCounter;
-        private double chargeTimeCounter;
+        /// <summary>
+        /// Whether the player is charging a charge of SlipTime.
+        /// </summary>
+        public bool IsChargingSlipTime { get; private set; }
 
         private void Start()
         {
             slipTimeCharges = maxSlipTimeCharges;
+            InSlipTime = false;
+            IsChargingSlipTime = false;
+            SlipTimeCounter = slipTimeDuration;
+            ChargeTimeCounter = slipTimeCooldownDuration;
```

Comment:
Note that the time counters now count down instead of up so that I can display their raw values in the GUI.